### PR TITLE
Color flag

### PR DESCRIPTION
**Pull request by pythondude325**
Add the standard `--color={always,auto,never}` CLI option. I might have missed some places where color is used.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -29,12 +32,20 @@ static ERRORS: AtomicUsize = AtomicUsize::new(0);
 static WARNINGS: AtomicUsize = AtomicUsize::new(0);
 
 const HELP: &str = concat!(
-    env!("CARGO_PKG_NAME"), " ", env!("RCC_GIT_REV"), "\n",
+    env!("CARGO_PKG_NAME"),
+    " ",
+    env!("RCC_GIT_REV"),
+    "\n",
```
Comment (jyn514) [inline src/main.rs:38]: This makes it look less like the output. Before one line of source corresponded to one line of help output.

File: src/main.rs [hunk 2]
```diff
@@ -62,7 +73,8 @@ OPTIONS:
 
 ARGS:
     <file>    The file to read C source from. \"-\" means stdin (use ./- to read a file called '-').
```
Comment (jyn514) [inline src/main.rs:75]: Because you added `r#"`, this now shows a literal backslash.

File: src/main.rs [hunk 3]
```diff
@@ -79,6 +91,47 @@ struct BinOpt {
...
+        match s {
+            "always" => Ok(ColorChoice::Always),
+            "auto" => Ok(ColorChoice::Auto),
+            "never" => Ok(ColorChoice::Never),
+            _ => Err("Invalid color choice"),
+        }
+    }
+}
+
+impl Into<color_backtrace::termcolor::ColorChoice> for ColorChoice {
```
Comment (jyn514) [inline src/main.rs:127]:
```suggestion
impl Into<termcolor::ColorChoice> for ColorChoice {
```

File: src/main.rs [hunk 4]
```diff
@@ -79,6 +91,47 @@ struct BinOpt {
...
+    fn into(self) -> termcolor::ColorChoice {
+        match self {
+            ColorChoice::Always => termcolor::ColorChoice::Always,
+            ColorChoice::Auto => termcolor::ColorChoice::Auto,
+            ColorChoice::Never => termcolor::ColorChoice::Never,
```
Comment (jyn514) [inline src/main.rs:132]: Personally I would `use termcolor::ColorChoice::*;` here, but it's up to you.

File: src/main.rs [hunk 5]
```diff
@@ -177,6 +232,16 @@ fn main() {
...
     };
+
+    #[cfg(debug_assertions)]
+    {
+        color_backtrace::install_with_settings(color_backtrace::Settings::new().output_stream(
+            Box::new(color_backtrace::termcolor::StandardStream::stderr(
+                opt.color.into(),
+            )),
+        ));
+    }
```
Comment (jyn514) [inline src/main.rs:243]: Is the outer block still necessary? It looks like you removed the `if`.

File: src/main.rs (hunk 1 above)
Comment (pythondude325) [reply src/main.rs:38]: I think rustfmt just did that. A little odd

File: src/main.rs (hunk 4 above)
Comment (pythondude325) [reply src/main.rs:132]: I usually prefer having paths

**Pull request merged**